src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -672,7 +672,12 @@ public override void SetCurrentDirectory(string fullPath)
 
         public override FileAttributes GetAttributes(string fullPath)
         {
-            return new FileInfo(fullPath, null).Attributes;
+            FileAttributes attributes = new FileInfo(fullPath, null).Attributes;
+
+            if (attributes == (FileAttributes)(-1))
+                FileSystemInfo.ThrowNotFound(fullPath);
```

Comment:
Just to confirm, it throws using the `DNFE/FNFE` pattern for `/path/diredoesntexist/filedoesntexit`, correct?